ext/include/opentelemetry/ext/zpages/zpages.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <chrono>
+
+#include "opentelemetry/ext/zpages/tracez_data_aggregator.h"
+#include "opentelemetry/ext/zpages/tracez_processor.h"
+#include "opentelemetry/ext/zpages/tracez_http_server.h"
+
+#include "opentelemetry/trace/provider.h"
+#include "opentelemetry/sdk/trace/tracer_provider.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+
+ /*
+  * Wrapper class that allows users to use zPages by calling this class constructor. Currently only
+  * has TraceZ, but other types of zPages can be added in the future
+  */
+class zPages {
+ public:
+  zPages()
+    : tracez_processor_(std::make_shared<opentelemetry::ext::zpages::TracezSpanProcessor>()),
+      tracez_provider_(opentelemetry::nostd::shared_ptr<opentelemetry::trace::TracerProvider>(
+        new opentelemetry::sdk::trace::TracerProvider(tracez_processor_))) {
+
+    // Set the global trace provider for a user to use, which is connected to our span processor
+    opentelemetry::trace::Provider::SetTracerProvider(tracez_provider_);
+
+    tracez_server_thread_ = std::thread(&zPages::RunTracezServer, this);
+    tracez_server_thread_.detach();
+
+    // Ensure zPages has time to setup, so the program doesn't crash
```

Comment:
Can you elaborate the comment -  why would it crash?

Reply:
I'm not sure why it crashes, as it just does if the thread doesn't sleep for at least 1 nanosecond. It's very odd behavior that I mentioned to Dustin, and we're thinking the maintainers may have better insights into this

Reply:
Interesting that 1 nanosecond is enough to avoid the crashing. Thanks for the info!